File: bin/jil.py
#!/usr/bin/env python

import sys
import re


sys.path.append('../lib')

from Database import *

jil = sys.stdin.readlines()

job = None

for line in jil:
    parameter, value = line.split(':', 1)
    
    parameter = parameter.strip()
    value = value.strip()
    
    if re.compile('#').search(parameter,1):
        pass
    
    elif 'update_job' == parameter:
        if job != None:
            commit_changes(job)
            
        job = Job().find(value)
        
        if job == None:
            print "Couldn't find job " + value + "\n"
            exit (-1)
            
        if ( job != None ):
            print "Updating job " + job.name + "\n"
        else:
            print "Update job: Couldn't find job with name " + value + "\n"
        
    elif 'insert_job' == parameter:
        if job != None:
            commit_changes(job)
        
        print "Inserting job " + value + "\n"
        job = Job()
        job.name = value
        
    elif 'delete_job' == parameter:
        print "Deleting job " + value + "\n"
        job = Job().find(value)
        delete_job(job)
        job = None
    
    elif job != None and parameter in ( 'std_in_file', 'std_err_file', 'std_out_file',
                                        'machine', 'start_mins', 'start_days', 'command'
                      ):
            print "Setting " + parameter + " to '" + value + "'\n"
            setattr(job, parameter, value)

            
if (job != None):
    commit_changes(job)

print "Complete.\n"

File: bin/jil.pl
#!/usr/bin/perl

# Copyright 2007 David Spadea
# All rights reserved.
#
# This code is release for use under the terms of the GNU General Public License, version 3.

use strict;
use warnings;

use Data::Dumper;
use Sys::Hostname;

use lib ("$ENV{ADSROOT}/lib");

use AdvancedScheduler qw(insert_job update_job delete_job);
use AdvancedScheduler::Database;
use AdvancedScheduler::JobDefinition;

my %callbacks = (
	insert_job => \&insert_job,
	update_job => \&update_job,
	delete_job => \&delete_job
);

my ($JIL, $exitcd);

while (<>)
{
	if ($_ =~ /insert_job|update_job|delete_job/)
	{
		if ($JIL)
		{
			$exitcd += &ProcessCommand($JIL);
			$JIL = undef;
		}
	}
	
	$JIL .= $_;
	
}

$exitcd += &ProcessCommand($JIL);

exit ($exitcd > 0 ? 1 : 0);

sub ProcessCommand
{
	my $def = AdvancedScheduler::JobDefinition->Parse($JIL)
		or die ('Unable to parse JIL.');
	
	# If namespace: is defined in JIL, use it. Otherwise,
	# consult the environment.
	$$def{namespace} ||= $ENV{ADSNAMESPACE};
	
	if ( $callbacks{$$def{ADSCOMMAND}} ) 
	{
		print "Issuing " . $$def{ADSCOMMAND} . " call for "
		      . join ("::", $$def{namespace}, $$def{name}) . "\n";
			      
		my $rc = $callbacks{$$def{ADSCOMMAND}}->($def);
		print "Database change ";
		print $rc ? "SUCCEEDED!\n" : "FAILED!\n";
		
		return ($rc ? 0 : 1);
	}
	else
	{
		warn ("Unknown command $$def{ADSCOMMAND}\n");
		return 2;
	}
}
